src/Databases/DatabaseReplicated.cpp

```diff
@@ -189,7 +215,17 @@ void DatabaseReplicated::propose(const ASTPtr & query) {
     current_zookeeper->createOrUpdate(zookeeper_path + "/last_entry", std::to_string(current_log_entry_n), zkutil::CreateMode::Persistent);
 
     lock->unlock();
-    // write to metastore the last entry?
+    saveState();
+}
+
+void DatabaseReplicated::createSnapshot() {
+    current_zookeeper->createAncestors(zookeeper_path + "/snapshot/huy");
```

Comment:
🤦‍♂️😁

Reply:
In order to create snapshot zookeeper nodes of kind `zookeeper_path + "/snapshot/" + table_name`, `zookeeper_path + "/snapshot/"` node must exist.
The little trick here does the job,
the path is specified in such a way so that the needed node will be created after the call.
The trick was needed cause `createAncestors` doesn't create a node itself.
However, since the trick is dirty by all means, it was removed and there was even a forced push so the commit history is erased.
Furthermore, the PR is WIP and `createSnapshot` might be completely changed later on.